pkg/store/gcs.go

```diff
@@ -65,6 +67,9 @@ func NewGCSStore(logger log.Logger, bucket *storage.BucketHandle, dir string) (*
 
 // Close the store.
 func (s *GCSStore) Close() (err error) {
+	s.mtx.RLock()
```

Comment:
Can we switch that to just `s.mtx.Lock()` (RWMutext to just Mutex?) 
We don't use the RW logic